os2phonebook_service/os2phonebook/integration.py

```diff
@@ -0,0 +1,702 @@
+from typing import List, Tuple
+from urllib.parse import urljoin
+from requests import Session, RequestException
+from os2phonebook.helpers import log_factory
+
+
+# Init loggign
+log = log_factory()
+
+
+class OS2MOImportClient:
+    """Client for importing org units and employees from the OS2MO service api.
+
+    This client uses requests.Session for HTTP(s) communication.
+
+    Attributes:
+        base_url (str): OS2MO service url, e.g. https://os2mo.magenta.dk
+        session (str): An instance of requests.Session
+        organisation_uuid (str): Main organisation identifier <uuid>
+        org_unit_map (dict): Hashmap for temporarily storing org units
+        employee_map (dict): Hashmap for temporarily storing employees
+
+    """
+
+    def __init__(self, base_url: str, session: Session = Session):
```

Comment:
What is the point of sending in the `Session` class as a parameter here?

Reply:
It was initially designed in this way to allow a test client to replace the Session class with a mocking class to avoid monkey patching.